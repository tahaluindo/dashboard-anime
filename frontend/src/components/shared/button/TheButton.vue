<template>
  <button :class="isActive ? active : notActive">
    <slot />
  </button>
</template>

<script lang="ts">
import { defineComponent } from "vue";

export default defineComponent({
  name: "TheButton",
  props: {
    active: {
      type: String,
      default: "",
    },
    notActive: {
      type: String,
      default: "",
    },
    isActive: {
      type: Boolean,
      default: false,
    },
  },
});
</script>
