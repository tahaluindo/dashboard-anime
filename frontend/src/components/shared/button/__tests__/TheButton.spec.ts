import { mount } from "@vue/test-utils";
import TheButton from "../TheButton.vue";

describe("TheButton.vue", () => {
  it("render a TheButton component with the passed ButtonLabel prop.", () => {
    const wrapper = mount(TheButton, {
      slots: {
        default: "Submit",
      },
    });

    const buttonLabel = wrapper.get("button").text();

    expect(buttonLabel).toBe("Submit");
    expect(wrapper.html()).toMatchSnapshot();
  });
});
