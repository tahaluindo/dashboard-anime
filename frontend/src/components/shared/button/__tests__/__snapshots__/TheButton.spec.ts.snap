// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheButton.vue render a TheButton component with the passed ButtonLabel prop. 1`] = `"<button class=\\"\\">Submit</button>"`;
